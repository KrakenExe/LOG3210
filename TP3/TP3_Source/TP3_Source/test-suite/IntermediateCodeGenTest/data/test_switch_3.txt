Declaration
 num a;
 num b;
 num c;
EndDeclaration

b = 42;
switch (b) {
    case 0 : a = 0;
    case 1 : a = 1;
    case c : a = 2;
    default : a = 3;
}
c = 5;
