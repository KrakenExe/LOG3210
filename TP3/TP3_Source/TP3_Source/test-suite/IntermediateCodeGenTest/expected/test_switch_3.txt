b = 42
_L1
goto _L3
_L4
a = 0
goto _L2
_L5
a = 1
goto _L2
_L6
a = 2
goto _L2
_L7
a = 3
goto _L2
_L3
if b == 0 goto _L4
if b == 1 goto _L5
if b == c goto _L6
goto _L7
_L2
c = 5
_L0
